java/cdk-custom-resources/src/test/java/sleeper/cdk/custom/TableDefinerLambdaIT.java

```diff
@@ -26,95 +31,312 @@
+            assertThat(streamTablePartitionTransactions(tableProperties)).isNotEmpty();
+            assertThat(listDataBucketObjectKeys())
+                    .extracting(FilenameUtils::getName)
+                    .containsExactly(
+                            FilenameUtils.getName(file.getFilename()),
+                            FilenameUtils.getName(file.getFilename()).replace("parquet", "sketches"));
+        }
+
+        @Test
+        void shouldFullyDeleteSpecifiedTableButNoOthers() throws Exception {
+            // Given
+            instanceProperties.set(RETAIN_DATA_AFTER_TABLE_REMOVAL, "false");
+            S3InstanceProperties.saveToS3(s3Client, instanceProperties);
+
+            TableProperties table1 = createTableProperties(uniqueIdAndName("test-table-1", "table-1"));
+            callLambda(CREATE, table1);
+
+            AllReferencesToAFile file1 = ingestRows(table1, List.of(
+                    new Row(Map.of("key1", 25L))));
+            assertThat(listDataBucketObjectKeys())
+                    .extracting(FilenameUtils::getName)
+                    .containsExactly(
+                            FilenameUtils.getName(file1.getFilename()),
+                            FilenameUtils.getName(file1.getFilename()).replace("parquet", "sketches"));
+
+            TableProperties table2 = createTableProperties(uniqueIdAndName("test-table-2", "table-2"));
+            callLambda(CREATE, table2);
+            AllReferencesToAFile file2 = ingestRows(table2, List.of(new Row(Map.of("key1", 25L))));
+
+            // When
+            callLambda(DELETE, table1);
+
+            // Then
+            assertThatThrownBy(() -> propertiesStore.loadByName(table1.get(TABLE_NAME)))
+                    .isInstanceOf(TableNotFoundException.class);
+            assertThat(streamTableFileTransactions(table1)).isEmpty();
+            assertThat(streamTablePartitionTransactions(table1)).isEmpty();
+            assertThat(propertiesStore.loadByName("table-2"))
+                    .isEqualTo(table2);
+            assertThat(streamTableFileTransactions(table2)).isNotEmpty();
+            assertThat(streamTablePartitionTransactions(table2)).isNotEmpty();
+            assertThat(listDataBucketObjectKeys())
+                    .extracting(FilenameUtils::getName)
+                    .containsExactly(
+                            FilenameUtils.getName(file2.getFilename()),
+                            FilenameUtils.getName(file2.getFilename()).replace("parquet", "sketches"));
+        }
+
+        @Test
+        void shouldFullyDeleteTableWhenSnapshotIsPresent() throws Exception {
+            // Given
+            instanceProperties.set(RETAIN_DATA_AFTER_TABLE_REMOVAL, "false");
+            S3InstanceProperties.saveToS3(s3Client, instanceProperties);
+
+            String tableName = tableProperties.get(TABLE_NAME);
+            tableProperties = createTableProperties(uniqueIdAndName(tableProperties.get(TABLE_ID), tableName));
```

Comment:
Similar to my previous comments, it doesn't look like there's a reason to recreate the same table properties object here?